frontend/test/resolution/testTypePropertyPrimitives.cpp

```diff
@@ -563,6 +567,92 @@ static void test11() {
   testPrimitive(tpg);
 }
 
+static void test12() {
+  Test tpg {
+    .testName = __FUNCTION__,
+    .isChplHomeRequired = false,    // TODO: True...
+    .prelude = R"""(
+            pragma "ignore noinit"
+            record r1 {}
+            record r2 { var x: int; var y: real; }
+            record r3 { var x: int; var y: real; var z: integral; }
+            record r4 { var x: r2; var y: int; }
+            record r5 {
+              proc deinit() {}
+            }
+            record r6 {
+              proc init=(rhs: r6) {}
+            }
+            record r7 {
+              operator=(lhs: r7, rhs: r7) {}
+            }
+            record r8 {}
+            operator=(lhs: r8, rhs: r8) {}
+            class c1 { var x: int; }
```

Comment:
Can you add to the test a record with compiler-generated `=` etc but that contains something non-POD?

Reply:
I have record `r3` which contains an `integral` field. Does that work?

Reply:
IMO the behavior around `isPOD(someGenericType)` is weird & I'd be looking for a case that doesn't fall into that.

Reply:
Ok, I've added a case for a record that contains an owned class.